src/super_gradients/training/utils/detection_utils.py

```diff
@@ -717,82 +740,378 @@ class NMS_Type(str, Enum):
+    :param recall_thresholds:   Recall thresholds used to compute MaP.
+    :param score_threshold:    Minimum confidence score to consider a prediction for the computation of
+                                    precision, recall and f1 (not MaP)
+    :param device:             Device
+
+    :return:
+        :precision, recall, ap, f1: Tensors of shape (n_class, nb_iou_thrs)
+    """
+    recall_thresholds = torch.linspace(0, 1, 101, device=device) if recall_thresholds is None else recall_thresholds
+
+    unique_classes = torch.unique(targets_cls)
+    n_class, nb_iou_thrs = len(unique_classes), preds_matched.shape[-1]
+
+    precision = torch.zeros((n_class, nb_iou_thrs), device=device)
+    recall = torch.zeros((n_class, nb_iou_thrs), device=device)
+    ap = torch.zeros((n_class, nb_iou_thrs), device=device)
+
+    for cls_i, cls in enumerate(unique_classes):
+        cls_preds_idx, cls_targets_idx = (preds_cls == cls), (targets_cls == cls)
+        cls_precision, cls_recall, cls_ap = compute_detection_metrics_per_cls(
+            preds_matched=preds_matched[cls_preds_idx],
+            preds_to_ignore=preds_to_ignore[cls_preds_idx],
+            preds_scores=preds_scores[cls_preds_idx],
+            n_targets=cls_targets_idx.sum(),
+            recall_thresholds=recall_thresholds,
+            score_threshold=score_threshold,
+            device=device
+        )
+        precision[cls_i, :] = cls_precision
+        recall[cls_i, :] = cls_recall
+        ap[cls_i, :] = cls_ap
+
+    f1 = 2 * precision * recall / (precision + recall + 1e-16)
+
+    return precision, recall, ap, f1, unique_classes
+
+
+def compute_detection_metrics_per_cls(
+        preds_matched: torch.Tensor,
+        preds_to_ignore: torch.Tensor,
+        preds_scores: torch.Tensor,
+        n_targets: int,
+        recall_thresholds: torch.Tensor,
+        score_threshold: float,
+        device: str,
+):
+    """
+    Compute the list of precision, recall and MaP of a given class for every recall IoU threshold.
+
+    :param preds_matched:      Tensor of shape (num_predictions, n_iou_thresholds)
+                                    True when prediction (i) is matched with a target with respect to the (j)th IoU threshold
+    :param preds_to_ignore     Tensor of shape (num_predictions, n_iou_thresholds)
+                                    True when prediction (i) is matched with a crowd target with respect to the (j)th IoU threshold
+    :param preds_scores:       Tensor of shape (num_predictions), confidence score for every prediction
+    :param n_targets:          Number of target boxes of this class
+    :param recall_thresholds:  Tensor of shape (max_n_rec_thresh) list of recall thresholds used to compute MaP
+    :param score_threshold:    Minimum confidence score to consider a prediction for the computation of
+                                    precision, recall and f1 (not MaP)
```

Comment:
There's no f1 as far as I can tell.

Reply:
Removed in bba66d102b960a772181b4da04ada6674b57bd48